### ISSUE TITLE
match_phrase does not work in highlight_query if stopwords are involved (FVH)

### ISSUE THREAD

**Issue by kori0129** [elastic/elasticsearch#26041]
Elasticsearch version: 5.2.0

Plugins installed: []

JVM version (java -version): 1.8.0_131

OS version (uname -a if on a Unix-like system):
inux 3.10.0-514.6.1.el7.x86_64 #1 SMP Wed Jan 18 13:06:36 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**:
**Action**: Highlight a field with a highlight_query, using match_phrase on a field with stopword filtering in Fast Vector Highlighter
**Expected Behaviour**: match_phrase handles stopwords the same way in highlight_query as in a normal query
**Actual Behaviour**: match_phrase fails to match in highlight_query with stopwords involved

**Steps to reproduce**:

 1.  Create index with a field with "term_vector" : "with_positions_offsets", to use Fast Vector Highlighter and a stopword analyser
```
PUT highlighttest
{
   "settings":{
      "analysis": {
          "analyzer": {
            "my_analyzer": {
              "type": "custom",
              "tokenizer": "standard",
              "filter": [ 
                "lowercase",
                "stopfilter"
              ]
            }
          },
          "filter": {
            "stopfilter": {
              "type":        "stop",
              "stopwords": [ "and", "the" ]  
            }
          }
          
       }
    },
   "mappings":{
      "doc":{
         "properties":{
            "body": {
               "type":"text",
               "analyzer":"my_analyzer",
               "term_vector" : "with_positions_offsets" 
            }
         }
      }
   }
}
```
  2. Index a document with stopwords in it
```
PUT highlighttest/doc/1
{
    "body": "I like the cake"
}
```

 3. Query the index with with a matchphrase in a highlight_query, while using stopwords in the searchterm

```
GET highlighttest/doc/_search
{
    "query": {
        "match_phrase": {
            "body":{
                "query":"like the cake"
            }
        }
    },
    "highlight": {
        "fields": {
             "body":{
                  "highlight_query": {
                        "match_phrase": {
                            "body":{
                                "query":"like the cake"
                            }
                        }
                   }
               }
         }
   }
}
```
the result is
```
{
   "took": 3,
   "timed_out": false,
   "_shards": {
      "total": 5,
      "successful": 5,
      "failed": 0
   },
   "hits": {
      "total": 1,
      "max_score": 0.5063205,
      "hits": [
         {
            "_index": "highlighttest",
            "_type": "doc",
            "_id": "1",
            "_score": 0.5063205,
            "_source": {
               "body": "I like the cake"
            }
         }
      ]
   }
}
```

As you can see, I'm using the exact same query in the normal query as in the highlight_query, and the article matches (so match_phrase matches), but it does not get highlighted.

Note: This works as expected with the plain highlighter. However, I cannot use plain highlighter for my product.

**Comment by javanna** [comment]
@jimczi would you have time to have a look at this?

**Comment by jimczi** [comment]
Yes the FVH does not work with stop words. This is a known bug in Lucene:
https://issues.apache.org/jira/browse/LUCENE-7551
The support for FVH has decreased in Lucene and efforts are made towards the unified highlighter now. This highlighter is available since 5.5 in es (and is the default in 6), works with fast vectors and handles phrase query with stop words efficiently:
https://www.elastic.co/guide/en/elasticsearch/reference/master/search-request-highlighting.html#unified-highlighter

I'll close this issue as won't fix, the resolution is to use the `unified` highlighter.
